uniffi/src/interface/types.rs

```diff
@@ -31,6 +31,12 @@ pub enum TypeReference {
     Float,
     Double,
     String,
+    // I don't like this too much since it's not a part of the IDL, and hopefully I can get rid of it
```

Comment:
I feel like I need to add more here for context 😞 

The asymmetry with how we handle strings in the `Rust` and `Kotlin` flow  has the following properties:
- When providing a string back from rust to kotlin, kotlin accepts the return value as a `Pointer`
- That pointer would need to be freed and kotlin would call a rust function to do that, which accepts a `Pointer`
- However, all other string accepting functions in the kotlin -> Rust flow, accept `String`

This causes a bit of a dilemma, we want to have a generic way to generate rust functions that accept strings, but we need to differentiate between whether that function is a `free` function. So this is why this variant exists. Now, this variant is only special for Kotlin, and if there is a nice way to get rid of it, I would be happy 😄

Reply:
Oh, interesting conundrum. It might be clearer if we named it `RawPointer` or something, but I'm happy to land with this and iterate.